src/functional/ffront/ast_passes/single_static_assign.py

```diff
@@ -108,7 +110,9 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AnnAssign:
         return node
 
     def visit_Name(self, node: ast.Name) -> ast.Name:
-        if node.id in self.state.name_counter:
+        if node.id.capitalize() in _BUILTINS:
```

Comment:
I don't understand why variable names are explicitly capitalized here instead of using lower case versions the list of builtins and compare by value without modifications. Could you elaborate on it?

Reply:
> Or even better in a third module where it is obvious that this should be the definitive list of names that the fieldview considers reserved for types.

I like this option better.

Additionally, I don't think the names need to be capitalized at all, even if they are used for dispatching in the `make_` methods of the TypeParser. I mean, `make_int32()` looks perfectly fine to me.

Reply:
I was thinking the same